src/azure-cli/azure/cli/command_modules/resource/_validators.py

```diff
@@ -32,9 +52,12 @@ def _validate_deployment_name(namespace):
 
 
 def process_deployment_create_namespace(namespace):
-    if bool(namespace.template_uri) == bool(namespace.template_file):
-        raise CLIError('incorrect usage: --template-file FILE | --template-uri URI')
-    _validate_deployment_name(namespace)
+    if bool(namespace.template_uri) == bool(namespace.template_file) == bool(namespace.template_spec):
+        raise CLIError('incorrect usage: --template-file FILE | --template-uri URI | --template-spec ID')
```

Comment:
How about specify that these three parameters cannot be passed in at the same time? Otherwise, it seems that the prompt is not clear enough.